cmd/zpool/zpool_main.c

```diff
@@ -1107,6 +1107,11 @@ zpool_do_create(int argc, char **argv)
 		}
 	}
 
+	/* Make new pools default to partition=raw */
+	if (add_prop_list_default(zpool_prop_to_name(ZPOOL_PROP_PARTITION),
+	    "raw", &props, B_TRUE))
+		goto errout;
```

Comment:
So I have a few questions, would using raw device cause it to not align to 1MB?
And the leaving space at the end is only done current on wholedisk=1 case?